utils.py

```diff
@@ -115,34 +117,90 @@ def apply_grounded_plan(self, state_grounded_atoms, plan):
-        return env.max_plan_step_reached
-
-    def get_shaped_reward(self, env, state, previous_state_grounded_atoms, next_state_grounded_atoms, plan):
+                if dynamic_reward_shaping is None:
+                    return i + env.max_plan_step_reached
+                elif dynamic_reward_shaping is "basic":
+                    if t == 0:
+                        return [i + env.max_plan_step_reached, i + env.max_plan_step_reached]
+                    return [(i + env.max_plan_step_reached) * self.phi_t(t), i + env.max_plan_step_reached]
+                elif dynamic_reward_shaping is "dist":
+                    return [ALPHA * self.dist_phi(env, state, plan[env.max_plan_step_reached]), i + env.max_plan_step_reached] #* self.phi_t(t)
+                else:
+                    raise NotImplementedError(f"{dynamic_reward_shaping} is not a valid dynamic reward shaping function")
+
+        if dynamic_reward_shaping is None:
+            return env.max_plan_step_reached
+        elif dynamic_reward_shaping is "basic":
+            if t == 0:
+                return [env.max_plan_step_reached, env.max_plan_step_reached]
+            return [(env.max_plan_step_reached) * self.phi_t(t), env.max_plan_step_reached]
+        elif dynamic_reward_shaping is "dist":
+            return [ALPHA * self.dist_phi(env, state, plan[env.max_plan_step_reached]), env.max_plan_step_reached] #* self.phi_t(t)
+        else:
+            raise NotImplementedError(f"{dynamic_reward_shaping} is not a valid dynamic reward shaping function")
+
+    def get_shaped_reward(self, env, prev_state, state, previous_state_grounded_atoms, next_state_grounded_atoms, plan, dynamic_reward_shaping):
         success = self.task_success_fn(env, state)
         reward = 1 if success else 0
 
-        prev_phi = self.phi(env, previous_state_grounded_atoms, plan)
+        if dynamic_reward_shaping is "dist":
+            prev_phi = self.phi(env, previous_state_grounded_atoms, plan, dynamic_reward_shaping, state=state)[1]
+        else:
+            prev_phi = self.phi(env, previous_state_grounded_atoms, plan, dynamic_reward_shaping)[1]
+        
         if env.max_plan_step_reached < prev_phi:
             env.max_plan_step_reached = prev_phi
-            
             # if max_plan_step_reached >= 9:
             #     print(env._t)
             #     print(dist_to_goal)
             #     import ipdb; ipdb.set_trace()
 
-        f = self.phi(env, next_state_grounded_atoms, plan) - self.phi(env, previous_state_grounded_atoms, plan)
+        if dynamic_reward_shaping is not None:
+            f = self.phi(env, next_state_grounded_atoms, plan, dynamic_reward_shaping, state=state)[0] - self.phi(env, previous_state_grounded_atoms, plan, dynamic_reward_shaping, state=prev_state)[0]
+        else:
+            f = self.phi(env, next_state_grounded_atoms, plan, dynamic_reward_shaping) - self.phi(env, previous_state_grounded_atoms, plan, dynamic_reward_shaping)
```

Comment:
oh i kinda see now; you need the `prev_phi` and the current phi; i think it'd be good to leave comments generally about what's going on so we don't get too confused over the next 2 days.